cmd/registry-replacer/main.go

```diff
@@ -509,3 +552,65 @@ func pruneReplacements(config *api.ReleaseBuildConfiguration, filter asDirective
 
 	return utilerrors.NewAggregate(errs)
 }
+
+func getPromotionTargetToDockerfileMapping(ocpBuildDataDir string, majorMinor ocpbuilddata.MajorMinor) (map[string]string, error) {
+	configs, err := ocpbuilddata.LoadImageConfigs(ocpBuildDataDir, majorMinor)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read image configs from ocp-build-data: %w", err)
+	}
+	result := map[string]string{}
+	for _, config := range configs {
+		result[config.PromotesTo()] = config.Dockerfile()
+	}
+	return result, nil
+}
+
+func updateDockerfilesToMatchOCPBuildData(
+	config *api.ReleaseBuildConfiguration,
+	promotionTargetToDockerfileMapping map[string]string,
+	majorMinorVersion string,
+	ignoredRepos sets.String,
+) {
+
+	// The tool only works for the current release
+	if config.Metadata.Branch != "master" {
+		return
+	}
+	if ignoredRepos.Has(config.Metadata.Org + "/" + config.Metadata.Repo) {
+		return
+	}
+
+	// Configs indexed by tag
+	promotedTags := map[string]api.ImageStreamTagReference{}
+	for _, promotedTag := range release.PromotedTags(config) {
+		if promotedTag.Namespace != "ocp" || promotedTag.Name != majorMinorVersion {
+			continue
+		}
+		promotedTags[promotedTag.Tag] = promotedTag
+	}
+	if len(promotedTags) == 0 {
+		return
+	}
+
+	for idx, image := range config.Images {
+		promotionTarget, ok := promotedTags[string(image.To)]
+		if !ok {
+			continue
+		}
+		stringifiedPromotionTarget := fmt.Sprintf("registry.svc.ci.openshift.org/%s/%s:%s", promotionTarget.Namespace, promotionTarget.Name, image.To)
+		dockerfilePath, ok := promotionTargetToDockerfileMapping[stringifiedPromotionTarget]
+		if !ok {
+			// TODO: Is "ocp build data doesn't know this" something we should handle?
+			continue
+		}
+		actualDockerFilePath := "Dockerfile"
+		if image.DockerfilePath != "" {
+			actualDockerFilePath = image.DockerfilePath
+		}
+		actualDockerFilePath = filepath.Join(image.ContextDir, actualDockerFilePath)
+		if dockerfilePath != actualDockerFilePath {
+			config.Images[idx].ContextDir = ""
```

Comment:
This sets the working directory for the build, so if their Dockerfile uses paths relative to this, and we change it, that could be breaking